sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -69,6 +74,18 @@
 COMMON_VALID_METRICS = sorted(
     set.intersection(*map(set, neighbors.VALID_METRICS.values()))
 )  # type: ignore
+
+# This can be extended to cover all distance metric objects, however that is
+# probably unnecessary and would slow down tests significantly.
+DISTANCE_METRIC_OBJS = []
```

Comment:
Can you explain what this is used for and add type annotations if possible?

Reply:
Replaced with a much simpler mechanism (possible due to https://github.com/scikit-learn/scikit-learn/pull/26471 😄)